tools/cli/commands/rsync.js

```diff
@@ -28,36 +28,45 @@ function escapeKey( key ) {
  * Stores the destination in the configstore.
  * Takes an optional alias arg.
  *
- * @param { string } dest - Destination path.
+ * @param { string } pluginDestPath - Destination path to plugins.
  * @param { string|false } alias - Alias key, if set.
  */
-function setRsyncDest( dest, alias = false ) {
-	const key = alias || dest;
-	rsyncConfigStore.set( escapeKey( key ), dest );
+function setRsyncDest( pluginDestPath, alias = false ) {
+	const key = alias || pluginDestPath;
+	rsyncConfigStore.set( escapeKey( key ), pluginDestPath );
 }
 
 /**
  * Entry point for the CLI.
+ * Prompts for the plugin.
+ * Prompts for the destination.
+ * Finds and uses the real wp plugin slug for dest.
  *
  * @param {object} argv - The argv for the command line.
  */
 export async function rsyncInit( argv ) {
 	argv = await promptForPlugin( argv );
 	argv = await promptForDest( argv );
 	const sourcePluginPath = projectDir( `plugins/${ argv.plugin }` ) + '/';
-	argv.dest = path.join( argv.dest, argv.plugin + '/' );
+	// Pull the actual plugin slug from composer.json.
+	const pluginComposerJson = fs.readFileSync(
+		projectDir( `plugins/${ argv.plugin }` + '/composer.json' )
+	);
+	const wpPluginSlug = JSON.parse( pluginComposerJson ).extra[ 'wp-plugin-slug' ];
```

Comment:
I note this isn't guaranteed to be set, although that's only likely for unpublished plugins like Jetpack Beta Tester or Debug Helper.

OTOH, I still think it'd probably be more straightforward to just have the user specify the destination directory instead of guessing what plugin name to append. That way things like wpcom's `wp-content/mu-plugins/jetpack-plugin/production/` don't need any special handling.

Reply:
What I liked about asking for /plugins path is that it's easier to reuse in a saved dest config or alias, and would be applicable to whatever plugin they're pushing there. 

In any case, my biggest concern right now is that seemingly the contents of the `jetpack_vendor/` symlinked packages are actually not being sent over. Only the empty directories

Reply:
Ah, it's failing because of the symlinks, the filter just points to the link and not all the files behind it.

Reverting to the explicit excludes just brings back all the problems from the prior commit. To do it right, we'll have to handle the symlinks properly when building the filter list. I'll look at writing code for that in a bit.